### PR TITLE
connection-manager: clean connection shutdown

### PR DESCRIPTION
**Pull request by coot**
- fixup! connection-manager: types & implementation
- connection-manager: clean connection shutdown
- diffusion: daProtocolIdleTimeout
- connection-manager: fix test
- connection-manager: updated documentation

### CODE REVIEW

File: ouroboros-network/src/Ouroboros/Network/PeerSelection/PeerStateActions.hs [hunk 1]
```diff
@@ -904,6 +904,9 @@ withPeerStateActions timeout
                     <$> awaitAllResults TokHot pchAppHandles
                     <*> awaitAllResults TokWarm pchAppHandles
                     <*> awaitAllResults TokEstablished pchAppHandles)
+      -- 'unregisterOutboundConnection' could only fail to demote the peer if
+      -- connection manager would simultanously protmoted it, but this is not
```
Comment (bolt12) [inline ouroboros-network/src/Ouroboros/Network/PeerSelection/PeerStateActions.hs:908]: simultanously protmoted -> simultaneously promoted

File: ouroboros-network/src/Ouroboros/Network/Diffusion.hs [hunk 2]
```diff
@@ -306,14 +306,21 @@ data DiffusionArguments m = DiffusionArguments {
       -- ^ run in initiator only mode
 
     , daProtocolIdleTimeout :: DiffTime
-      -- ^ Timeout which starts once all responder protocols are idle. If the
-      -- responders stay idle for duration of the timeout, the connection will
-      -- be demoted, if it wasn't used by the p2p-governor it will be closed.
+      -- ^ Configures timeout which starts when initiator and responder
+      -- mini-protocols bacme idle.  If the initiators \/ responders stay idle
```
Comment (bolt12) [inline ouroboros-network/src/Ouroboros/Network/Diffusion.hs:310]: bacme -> become

**Comment by bolt12** [comment]
CI is not building, but LGTM so far!

**Comment by bolt12** [comment]
Is db191417f8ca763d2bb9a6d18da8aacaef41c6c2 supposed to be fixed-up?

**Comment by coot** [comment]
> Is db19141 supposed to be fixed-up?
It targets a commit that is deep in the `p2p-master` branch.

**Review by bolt12 — APPROVED** [review]
LGTM!

File: ouroboros-network-framework/src/Ouroboros/Network/InboundGovernor.hs [hunk 3]
```diff
@@ -114,6 +116,10 @@ inboundGovernor tracer serverControlChannel inboundIdleTimeout
                         tracer
                         (igsCountersCache state)
                         (inboundGovernorCounters state)
+      traceWith tracer $ TrRemoteState $
+            mkRemoteSt . csRemoteState
+        <$> igsConnections state
+
```
Comment (bolt12) [inline ouroboros-network-framework/src/Ouroboros/Network/InboundGovernor.hs:122]: Should this be traced with `Cache`?

Comment (coot) [reply ouroboros-network-framework/src/Ouroboros/Network/InboundGovernor.hs:122]: This is a debug tracer, so it's better to not cache it.

**Pull request merged**